packages/app-page-builder/src/editor/plugins/elementSettings/save/SaveDialog.tsx

```diff
@@ -66,7 +66,7 @@ const SaveDialog = (props: Props) => {
         setTimeout(async () => {
             setPbElement((await getElementTree({ element })) as PbElement);
         });
-    });
+    }, []);
```

Comment:
Should we rather use `element.id` as the cache key? Could be more precise.

Also, this whole `useEffect` block was added probably with the improved block previews PR, but I guess I missed this. Can we add a short comment explaining why this is needed?